test/GPv2TradeExecution.test.ts

```diff
@@ -1,315 +0,0 @@
-
-  let tradeExecution: Contract;
-
-  beforeEach(async () => {
-    const GPv2TradeExecution = await ethers.getContractFactory(
-      "GPv2TradeExecutionTestInterface",
-      recipient,
-    );
-
-    tradeExecution = await GPv2TradeExecution.deploy();
-  });
-
-  describe("transferSellAmountToRecipient", () => {
-    const withoutBuy = {
-      receiver: ethers.constants.AddressZero,
-      buyToken: ethers.constants.AddressZero,
-      buyAmount: ethers.constants.Zero,
-    };
-
-    it("should transfer sell amount to recipient", async () => {
-      const amount = ethers.utils.parseEther("13.37");
-
-      const sellToken = await waffle.deployMockContract(deployer, IERC20.abi);
-      await sellToken.mock.transferFrom
-        .withArgs(traders[0].address, recipient.address, amount)
-        .returns(true);
-
-      await expect(
-        tradeExecution.transferSellAmountToRecipientTest(
-          {
-            owner: traders[0].address,
-            sellToken: sellToken.address,
-            sellAmount: amount,
-            ...withoutBuy,
-          },
-          recipient.address,
-        ),
-      ).to.not.be.reverted;
-    });
-
-    it("should revert on failed ERC20 transfers", async () => {
-      const amount = ethers.utils.parseEther("4.2");
-
-      const sellToken = await waffle.deployMockContract(deployer, IERC20.abi);
-      await sellToken.mock.transferFrom
-        .withArgs(traders[0].address, recipient.address, amount)
-        .revertsWithReason("test error");
-
-      await expect(
-        tradeExecution.transferSellAmountToRecipientTest(
-          {
-            owner: traders[0].address,
-            sellToken: sellToken.address,
-            sellAmount: amount,
-            ...withoutBuy,
-          },
-          recipient.address,
-        ),
-      ).to.be.revertedWith("test error");
-    });
-
-    it("does not revert when transfering a token with no contract at its address", async () => {
-      await expect(
-        tradeExecution.transferSellAmountToRecipientTest(
-          {
-            owner: traders[0].address,
-            sellToken: traders[1].address,
-            sellAmount: ethers.utils.parseEther("1.0"),
-            ...withoutBuy,
-          },
-          recipient.address,
-        ),
-      ).not.to.be.reverted;
-    });
-
-    it("reverts when mistakenly trying to sell ETH using the marker buy Ether address", async () => {
-      await expect(
-        tradeExecution.transferSellAmountToRecipientTest(
-          {
-            owner: traders[0].address,
-            sellToken: BUY_ETH_ADDRESS,
-            sellAmount: ethers.utils.parseEther("1.0"),
-            ...withoutBuy,
-          },
-          recipient.address,
-        ),
-      ).to.be.revertedWith("GPv2: cannot transfer native ETH");
-    });
-
-    describe("Non-Standard ERC20 Tokens", () => {
-      it("should not revert when ERC20 transfer has no return data", async () => {
```

Comment:
Do these tests still appear in the updated code somewhere?

Reply:
No, I thought it was redundant since we already test this in `GPv2SafeERC20.test.ts`:

https://github.com/gnosis/gp-v2-contracts/blob/37f5a84b3664f76112f5e0e4a8fc31feb6f540c7/test/GPv2SafeERC20.test.ts#L52

In general, I'm not a huge fan of duplicating tests when they share a code path. On the other hand, I do see the argument that we may "forget" to wrap a call in a "safe ERC20" call. In that case, I believe we would be better served with an E2E tests that settlement works with some well-known non-standard ERC20 token implementation.

Reply:
I like the idea of adding it back as an end to end test. :+1:

Reply:
Implemented in #515